### ISSUE TITLE
R script title and author not used correctly

### ISSUE THREAD

**Issue by andyreagan** [yihui/knitr#1869]
With an R script called `example_notebook.R` as 
```
#' ---
#' title: "Crop Analysis Q3 2013"
#' author: "John Smith"
#' date: "May 3rd, 2014"
#' output:
#'   bookdown::html_document2:
#'     toc: TRUE
#' ---
```
when I use

```
rmarkdown::render('example_notebook.R')
```

the title, author, and date are not picked up (I get the title as `example_notebook.R`, the author as my $USER, and the date as today).

It works if I first spin and then knit:

```
knitr::spin('example_notebook.R', knit=FALSE)
rmarkdown::render('example_notebook.Rmd')
```

Here are my versions of things:
```R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.5

Locale: en_US.UTF-8 / en_US.UTF-8 / en_US.UTF-8 / C / en_US.UTF-8 / en_US.UTF-8

Package version:
  base64enc_0.1.3 digest_0.6.25   evaluate_0.14   glue_1.4.1     
  graphics_4.0.0  grDevices_4.0.0 highr_0.8       htmltools_0.4.0
  jsonlite_1.6.1  knitr_1.28      magrittr_1.5    markdown_1.1   
  methods_4.0.0   mime_0.9        Rcpp_1.0.4.6    rlang_0.4.6    
  rmarkdown_2.3   stats_4.0.0     stringi_1.4.6   stringr_1.4.0  
  tinytex_0.23    tools_4.0.0     utils_4.0.0     xfun_0.14      
  yaml_2.2.1     

Pandoc version: 2.9.2.1
```

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('knitr')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('yihui/knitr')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by andyreagan** [comment]
Well, restarting R seems to have fixed the issue. I ran this MWE again, and I can't see to break it.

**Comment by yihui** [comment]
This should be the same bug as https://github.com/rstudio/rmarkdown/issues/1740, which was fixed in [**rmarkdown** 2.2](https://github.com/rstudio/rmarkdown/releases/tag/v2.2).
